homeassistant/components/media_player/androidtv.py

```diff
@@ -0,0 +1,413 @@
+    DOMAIN, MediaPlayerDevice, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE, SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_STOP, SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP)
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_DEVICES, CONF_NAME, CONF_PORT,
+    CONF_SCAN_INTERVAL, STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_OFF)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['pure-python-adb>=0.1.5.dev0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Android'
+DEFAULT_PORT = 5555
+SCAN_INTERVAL = timedelta(seconds=10)
+
+SUPPORT_ANDROIDTV = (SUPPORT_NEXT_TRACK | SUPPORT_PAUSE |
+                     SUPPORT_PLAY | SUPPORT_PREVIOUS_TRACK |
+                     SUPPORT_TURN_OFF | SUPPORT_TURN_ON |
+                     SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP |
+                     SUPPORT_STOP)
+
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA}, })
+
+KNOWN_APPS = {
+    "dream": "Screensaver",
+    "kodi": "Kodi",
+    "netfilx": "Netflix",
+    "plex": "Plex",
+    "spotify": "Spotify",
+    "tvlauncher": "Homescreen",
+    "youtube": "Youtube"
+}
+
+ACTIONS = {
+    "back": "4",
+    "blue": "186",
+    "component1": "249",
+    "component2": "250",
+    "composite1": "247",
+    "composite2": "248",
+    "down": "20",
+    "end": "123",
+    "enter": "66",
+    "green": "184",
+    "hdmi1": "243",
+    "hdmi2": "244",
+    "hdmi3": "245",
+    "hdmi4": "246",
+    "home": "3",
+    "input": "178",
+    "left": "21",
+    "menu": "82",
+    "move_home": "122",
+    "mute": "164",
+    "pairing": "225",
+    "power": "26",
+    "resume": "224",
+    "right": "22",
+    "sat": "237",
+    "search": "84",
+    "settings": "176",
+    "sleep": "223",
+    "suspend": "276",
+    "sysdown": "281",
+    "sysleft": "282",
+    "sysright": "283",
+    "sysup": "280",
+    "text": "233",
+    "top": "122",
+    "up": "19",
+    "vga": "251",
+    "voldown": "25",
+    "volup": "24",
+    "yellow": "185"
+}
+
+DATA_KEY = '{}.androidtv'.format(DOMAIN)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the androidtv platform."""
+
+    if DATA_KEY not in hass.data:
+        hass.data[DATA_KEY] = {}
+
+    from adb.client import Client as AdbClient
+    client = AdbClient(host="127.0.0.1", port=5037)
+
+    androidtvs = []
+    for host, device_config in config[CONF_DEVICES].items():
+        name = device_config[CONF_NAME]
+        port = str(device_config[CONF_PORT])
```

Comment:
Move any value modifications to the config schema.

Reply:
`vol.Optional(str(CONF_PORT), default=DEFAULT_PORT): cv.port,` in the schema doesn't work, it's still not a string, am I missing something ?